### PR TITLE
Use Crowdin’s v2 API

### PR DESCRIPTION
**Pull request by rppld**
Uses [`@crowdin/crowdin-api-client`](https://github.com/crowdin/crowdin-api-client-js) to interact with Crowdin’s API and adds a new command `delete-branch`.

I guess it can be improved in some places, so this should be a starting point for discussion. Didn’t update `@formatjs/cli` in this PR, as it introduced a few breaking changes.

### CODE REVIEW

File: src/download.ts [hunk 1]
```diff
@@ -1,79 +1,79 @@
...
-  log.info("", `Writing translations to: ${config.TRANSLATIONS_DIR}`);
+  log.info(`Writing translations to: ${chalk.bold(config.TRANSLATIONS_DIR)}`);
 
   await Promise.all(
     config.CROWDIN_LANGUAGES.map((language, i) => {
-      const translations = exportFileResponses[i].data;
-      if (translations.error) {
+      const response = exportFileResponses[i];
+
+      if (!response || response.data.error) {
```
Comment (CosmaTrix) [inline src/download.ts:51]: This was causing issues in v1 already. If there is a string with an `error` key the scripts will stop here. Maybe it isn't for the context of this PR, so we should work on a follow-up version that handles HTTP errors in a different way (throw an error instead of resolving with an `error` key that could conflict with a translation string).

File: src/lib/crowdin.ts [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+  token: CROWDIN_PERSONAL_ACCESS_TOKEN,
+};
+
+const {
+  translationsApi,
+  sourceFilesApi,
+  uploadStorageApi,
+} = new CrowdinApiClient(credentials);
+
+export function unwrapValidationErrorResponse(
```
Comment (CosmaTrix) [inline src/lib/crowdin.ts:28]: Please use arrow functions like in the other cases.

File: src/utils/logging.ts [hunk 3]
```diff
@@ -1,19 +1,11 @@
-import chalk from "chalk";
-
-// eslint-disable-next-line
+// eslint-disable-next-line no-console, @typescript-eslint/no-empty-function
 const log = process.env.NODE_ENV !== "test" ? console.log : () => {};
```
Comment (CosmaTrix) [inline src/utils/logging.ts:2]: Little trick: You can use `() => null` or `() => undefined` if you want to get rid of the ESLint suppression comment.

**Review by CosmaTrix — CHANGES_REQUESTED** [review]

File: src/config.ts [hunk 4]
```diff
@@ -1,61 +1,44 @@
...
-
-require("dotenv").config();
+dotenv.config();
 
-if (
-  !process.env.CROWDIN_KEY ||
-  !process.env.CROWDIN_PROJECT_NAME ||
-  !process.env.CROWDIN_LANGUAGES
-) {
+if (!process.env.CROWDIN_LANGUAGES) {
```
Comment (el-angel) [inline src/config.ts:9]: can you add a check for the other variables as well?

Comment (rppld) [reply src/config.ts:9]: Ahh, good you spotted this! Removed the old vars, but forgot to add the new ones 😅

File: src/download.ts (hunk 1 above)
Comment (el-angel) [reply src/download.ts:51]: maybe we can also check the returned statuscode?

**Comment by el-angel** [comment]
Very happy with this PR @rppld!  Especially the delete-branch feature will help us so much in the feature when we're able to implement that in pipelines 🤩👏

**Comment by rppld** [comment]
@el-angel nice, yes! Couple more changes coming in from Dami soon and the tests need a bit of fixing. But should be ready soon!

File: src/download.ts (hunk 1 above)
Comment (rppld) [reply src/download.ts]: Yep, indeed. Checking for the status code now. Tried to reproduce all edge-cases in the mobile app project and seems resolved.

File: src/collect.ts [hunk 5]
```diff
@@ -33,42 +25,5 @@ export default async (glob: string) => {
...
-  const sortedMessages = sortByKey(messages);
-
-  log.info("", `${Object.keys(sortedMessages).length} translations found.`);
-
-  const newTranslations = JSON.stringify(sortedMessages, null, 2);
-
-  log.info("", "Writing new translations source file.");
-  fs.writeFileSync(config.TRANSLATIONS_FILE, newTranslations);
-
   return;
```
Comment (CosmaTrix) [inline src/collect.ts:28]: This `return` can be removed, can't it?

File: src/download.ts [hunk 6]
```diff
@@ -1,79 +1,79 @@
...
-  log.info("", `Writing translations to: ${config.TRANSLATIONS_DIR}`);
+  log.info(`Writing translations to: ${chalk.bold(config.TRANSLATIONS_DIR)}`);
 
   await Promise.all(
     config.CROWDIN_LANGUAGES.map((language, i) => {
-      const translations = exportFileResponses[i].data;
-      if (translations.error) {
+      const response = exportFileResponses[i];
+
+      if (!response || response.status !== 200) {
```
Comment (CosmaTrix) [inline src/download.ts:51]: If possible, check if the response is `ok` or outside the 200-299 range.

**Review by CosmaTrix — CHANGES_REQUESTED** [review]

File: src/download.ts (hunk 6 above)
Comment (rppld) [reply src/download.ts]: Ah yes, wanted to do that originally, but `ok` is not available on the [Axios response object](https://github.com/axios/axios#response-schema). Changed it to check for `statusText` instead ✌️

**Review by CosmaTrix — APPROVED** [review]
